odxtools/diagdatadictionaryspec.py

```diff
@@ -30,32 +30,16 @@ def _construct_named_item_list(iterable):
 
 @dataclass
 class DiagDataDictionarySpec:
-    dtc_dops: NamedItemList[DtcDop] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    data_object_props: NamedItemList[DataObjectProperty] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    structures: NamedItemList[BasicStructure] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    end_of_pdu_fields: NamedItemList[EndOfPduField] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    tables: NamedItemList[Table] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    env_data_descs: NamedItemList[EnvironmentDataDescription] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    env_datas: NamedItemList[EnvironmentData] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    muxs: NamedItemList[Multiplexer] = field(
-        default_factory=lambda: _construct_named_item_list([])
-    )
-    unit_spec: Optional[UnitSpec] = None
-    sdgs: List[SpecialDataGroup] = field(default_factory=list)
+    dtc_dops: NamedItemList[DtcDop]
+    data_object_props: NamedItemList[DataObjectProperty]
+    structures: NamedItemList[BasicStructure]
+    end_of_pdu_fields: NamedItemList[EndOfPduField]
+    tables: NamedItemList[Table]
+    env_data_descs: NamedItemList[EnvironmentDataDescription]
+    env_datas: NamedItemList[EnvironmentData]
+    muxs: NamedItemList[Multiplexer]
+    unit_spec: Optional[UnitSpec]
+    sdgs: List[SpecialDataGroup]
```

Comment:
Still needed?
````
# The attributes are already initialized as (most likely normal) lists.
# Convert them to NamedItemLists if necessary
```

Reply:
actually no: now NamedItemLists need to be provided to the constructor, i.e., no need to convert them in post anymore. Since this is IMO a bit out of this PR's scope, I'll fix that in the next one...